layers/+filetree/treemacs/funcs.el

```diff
@@ -21,8 +21,9 @@
       (treemacs-do-add-project-to-workspace path name)
       (treemacs-select-window))))
 
-(defun spacemacs/treemacs-toggle-locked-width-off ()
-  "Unlock the manual resizing of the treemacs window."
+(defun spacemacs/treemacs-setup-width-lock ()
+  "Setup the width lock of treemacs buffer based on
+`treemacs-lock-width'."
   (interactive)
-  (when treemacs--width-is-locked
+  (unless (eq treemacs--width-is-locked treemacs-lock-width)
```

Comment:
Mmmmmmh this will make the docs non-accurate, since a simply non-nil value won't always do it. It should be

```
(unless treemacs-locked-width
  (when treemacs--width-is-locked
    (treemacs-toggle-fixed-width)))
```
It may be longer, but at least it makes clear what the intent is.

Reply:
Good point, there might be some shit instead of `t`. I'd use it like this:

```
(unless (eq (not treemacs--width-is-locked) (not treemacs-lock-width))
  ..)
```

I don't know how to compare two 'boolean' 'objects' 😭 😸

Reply:
Just, simply 
```
(unless treemacs-locked-width
  (when treemacs--width-is-locked
    (treemacs-toggle-fixed-width)))
```
`Unless they share the same thruthyness` is clever but I think it also hides the intent behind it. It makes it hard to read because you also have to be clever (or think hard) to read it.

Reply:
Just, simply 
```
(unless treemacs-locked-width
  (when treemacs--width-is-locked
    (treemacs-toggle-fixed-width)))
```
`Unless they share the same thruthyness` is clever but I think it also hides the intent behind it. It makes it hard to read because you also have to be clever (or think hard) to read it.

Reply:
It holds true iff set `treemacs-locked-width` to `nil` and treemacs is currently locked. But my point was about the situation when treemacs is not locked and the user wants to lock it. I know that by default treemacs is locked, but I would not place my stake on implicit* default value from third-party 😸 One day they change behaviour and you have to come back and change this code. Being bullet-proof against changes outside of the project is a good thing I think :D 

* implicit here means that we do not use any real variable here, we just /know/ that it holds 😸

Reply:
I see what you mean, did'nt think of it like that.

`(unless (eq (not treemacs--width-is-locked) (not treemacs-lock-width)))` would be equivalent to `(when (xor treemacs-lock-width treemacs--width-is-locked))` if I'm not mistaken.